src/Microsoft.AspNetCore.SignalR.Core/HubEndPoint.cs

```diff
@@ -400,6 +400,11 @@ private static bool IsChannel(Type type, out Type payloadType)
                     await SendMessageAsync(connection, new StreamItemMessage(invocationId, enumerator.Current));
                 }
             }
+            catch (ChannelClosedException ex)
```

Comment:
:( Not a huge fan of this, but I don't see a better way to do this via IAsyncEnumerator. We could use the channel APIs directly I guess, and use `WaitToRead` to detect this case (and then `.Completion` to get the exception).